src/vm/database/sqlite.rs

```diff
@@ -297,22 +324,88 @@ impl <'a> ContractDatabase <'a> {
     }
 
     pub fn get_simmed_block_height(&self) -> Result<i128> {
-        let block_height: (i128) =
+        let block_height: (i64) =
```

Comment:
This is unsafe-ish.  You should really pull out the `i128` blob from the DB, since that's what you're returning anyway.

Reply:
These have all be updated, following our convo. All casts are now checked beforehand.